src/main/java/core/basesyntax/MainApp.java

```diff
@@ -1,5 +1,15 @@
 package core.basesyntax;
 
 public class MainApp {
+    public static void main(String[] args){
+        Truck truck = new Truck();
+        Bulldozer bulldozer = new Bulldozer();
+        Excavator excavator = new Excavator();
```

Comment:
Use abstract references instead of specific ones. Instead of declaring 'Truck', 'Bulldozer', and 'Excavator' specifically, use the 'Machine' type for the references. This will allow you to take full advantage of polymorphism.